### PR TITLE
Add cpp examples

### PR DESCRIPTION
**Pull request by dyashuni**
Add cpp examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,7 +123,7 @@ Properties of `hnswlib.Index` that support reading and writing:
         
   
 #### Python bindings examples
-[See more examples here](examples/EXAMPLES.md)
+[See more examples here](examples/python/EXAMPLES.md)
```
Comment (yurymalkov) [inline README.md:126]: Maybe give a one-two word description of the examples

Comment (dyashuni) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -123,7 +123,12 @@ Properties of `hnswlib.Index` that support reading and writing:
         
   
 #### Python bindings examples
-[See more examples here](examples/EXAMPLES.md)
+[See more examples here](examples/python/EXAMPLES.md):
+* creating index, inserting elements, searching, serialization/deserialization
+* filtering during the search
+* delete elements and reusing the memory of the deleted elements when new elements are being added
```
Comment (yurymalkov) [inline README.md:129]:
```suggestion
* deleting the elements and reusing the memory of the deleted elements for newly added elements
```

File: README.md [hunk 3]
```diff
@@ -123,7 +123,12 @@ Properties of `hnswlib.Index` that support reading and writing:
         
   
 #### Python bindings examples
-[See more examples here](examples/EXAMPLES.md)
+[See more examples here](examples/python/EXAMPLES.md):
+* creating index, inserting elements, searching, serialization/deserialization
+* filtering during the search
```
Comment (yurymalkov) [inline README.md:128]:
```suggestion
* filtering during the search with a boolean function 
```

File: README.md [hunk 4]
```diff
@@ -230,6 +235,14 @@ labels, distances = p.knn_query(data, k=1)
 print("Recall for two batches:", np.mean(labels.reshape(-1) == np.arange(len(data))), "\n")
 ```
 
+#### C++ examples
+[See examples here](examples/cpp/EXAMPLES.md):
+* creating index, inserting elements, searching, serialization/deserialization
+* filtering during the search
```
Comment (yurymalkov) [inline README.md:241]:
```suggestion
* filtering during the search with a boolean function
```

File: README.md [hunk 5]
```diff
@@ -230,6 +235,14 @@ labels, distances = p.knn_query(data, k=1)
...
+* delete elements and reusing the memory of the deleted elements when new elements are being added
```
Comment (yurymalkov) [inline README.md:242]:
```suggestion
* deleting the elements and reusing the memory of the deleted elements for newly added elements
```

File: examples/cpp/EXAMPLES.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    recall = (float)correct / max_elements;
+    std::cout << "Recall of deserialized index: " << recall << "\n";
+
+    delete[] data;
+    delete alg_hnsw;
+    return 0;
+}
+```
+
+An example with a filter during the search:
```
Comment (yurymalkov) [inline examples/cpp/EXAMPLES.md:65]:
```suggestion
An example of filtering with a boolean function during the search:
```

File: examples/cpp/example_mt_search.cpp [hunk 7]
```diff
@@ -0,0 +1,105 @@
+#include "../../hnswlib/hnswlib.h"
+#include <thread>
+
+
+// Multithreaded executor
+template<class Function>
+inline void ParallelFor(size_t start, size_t end, size_t numThreads, Function fn) {
```
Comment (yurymalkov) [inline examples/cpp/example_mt_search.cpp:7]:
```suggestion
// The helper function copied from python_bindings/bindings.cpp (and that itself if copied from nmslib)
// An alternative is using #pragme omp parallel for or any other C++ threading
template<class Function>
inline void ParallelFor(size_t start, size_t end, size_t numThreads, Function fn) {
```

**Review by yurymalkov — APPROVED** [review]
A couple suggestions, otherwise seems to be good! Thanks!

File: examples/cpp/example_mt_search.cpp (hunk 7 above)
Comment (dyashuni) [reply examples/cpp/example_mt_search.cpp:9]: done

**Pull request merged**